tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -524,9 +585,15 @@ private[optimizer] abstract class OptimizerCore(
           pretransformExpr(expr) { texpr =>
             texpr.tpe match {
               case RefinedType(base: ReferenceType, true, false) =>
+                val actualBase = base match {
+                  case ClassType(LongImpl.RuntimeLongClass) =>
+                    ClassType(Definitions.BoxedLongClass)
+                  case _ =>
+                    base
+                }
```

Comment:
It is already in desugar, for the run-time semantics. (Actually in `scalajsenv.js`) Here it is for constant-folding of `GetClass`.